matching-service/controllers/MatchingController.js

```diff
@@ -1,322 +1,110 @@
-
-            var res = { 
-                status: responseStatus.SUCCESS,
-                data: {
-                    interviewId: interview.interviewId,
-                    partnerEmail: partnerResult.email,
-                    difficulty: difficulty,
-                    secondsLeft: 3600
-                }
-            };
-            
-            socket.emit("matchSuccess", res);
-            io.emit("matchSuccess", "success match");
-            return res;
-        } catch (err) {
-            clearInterval(intervalId);
-            var res = { 
-                status: responseStatus.ERROR, 
-                message: mongoErrMsgs.writeError(err)
-            };
-            socket.emit("matchFailed", res);
-            return res;
-        }
-    }, 5000); 
-
+  var res = {
+    status: responseStatus.SUCCESS,
+    data: {
+      message: logMsgs.SERVICE_HEALTHY,
+    },
+  };
+  return res;
 }
 
-// Get number of current ongoing interviews
-async function interviewsCount() {
-    try {
-        const numInterviews = await Interview.countDocuments({});
-        var res = {
-            status: responseStatus.SUCCESS,
-            data: {
-                count: numInterviews
-            }
-        };
-        return res;
-    } catch (err) {
-        var res = {
-            status: responseStatus.ERROR,
-            message: mongoErrMsgs.readError(err)
-        };
-        return res;
-    }
-}
-
-// Get interview of user (if it exist)
-async function getInterview(email) {
-    try {
-        const interview = await Interview.findOne({
-            $or: [
-                { email1: email },
-                { email2: email }
-            ]
-        });
-
-        if (!interview) {
-            var res = {
-                status: responseStatus.NOT_FOUND,
-                message: clientErrMsgs.NO_INTERVIEW_FOUND_ERR
-            }
-            return res;
-        }
-
-        var partnerEmail = interview.email1;
-        if (partnerEmail == email) {
-            partnerEmail = interview.email2;
-        }
-
-        var secondsPassed = Math.floor((new Date().getTime() - interview.timeCreated.getTime()) / 1000);
-        const secondsLeft = 3600 - secondsPassed;
-
-        var res = {
-            status: responseStatus.SUCCESS,
-            data: {
-                partnerEmail: partnerEmail,
-                interviewId: interview.interviewId,
-                secondsLeft: secondsLeft
-            }
-        };
-
-        return res;
-    } catch (err) {
-        var res = {
-            status: responseStatus.ERROR,
-            message: mongoErrMsgs.readError(err)
-        };
-        return res;
-    }
+async function searchMatch(socket, io, email, difficulty, jwtToken, userId) {
+  var authRes = authJwt.verifyToken(jwtToken, userId, socket);
+
+  // Not authorised
+  if (authRes.status === responseStatus.UNAUTHORIZED) {
+    socket.emit("unauthorized", authRes);
+    console.log("Unauthorized");
+    return;
+  }
```

Comment:
I see that this only checks for `UNAUTHORIZED` does it mean for `BAD_REQUEST` the socket does not emit any error event?

Reply:
i will remove the BAD_REQUEST and only have unauthorized